crates/settlement-clients/starknet/src/lib.rs

```diff
@@ -128,20 +128,26 @@ impl SettlementClient for StarknetSettlementClient {
 
     /// Should verify the inclusion of a tx in the settlement layer
     async fn verify_tx_inclusion(&self, tx_hash: &str) -> Result<SettlementVerificationStatus> {
-        let tx_hash = FieldElement::from_hex_be(tx_hash)?;
+        let tx_hash = Felt::from_hex(tx_hash)?;
         let tx_receipt = self.account.provider().get_transaction_receipt(tx_hash).await?;
-        match tx_receipt {
-            MaybePendingTransactionReceipt::Receipt(tx) => match tx.execution_result() {
-                ExecutionResult::Succeeded => Ok(SettlementVerificationStatus::Verified),
-                ExecutionResult::Reverted { reason } => {
-                    Ok(SettlementVerificationStatus::Rejected(format!("Tx {} has been reverted: {}", tx_hash, reason)))
-                }
-            },
-            MaybePendingTransactionReceipt::PendingReceipt(tx) => match tx.execution_result() {
-                ExecutionResult::Succeeded => Ok(SettlementVerificationStatus::Pending),
-                ExecutionResult::Reverted { reason } => Ok(SettlementVerificationStatus::Rejected(format!(
+        let execution_result = tx_receipt.receipt.execution_result();
+        let status = execution_result.status();
+
+        match tx_receipt.block.is_pending() {
+            true => match status {
+                TransactionExecutionStatus::Succeeded => Ok(SettlementVerificationStatus::Pending),
+                TransactionExecutionStatus::Reverted => Ok(SettlementVerificationStatus::Rejected(format!(
                     "Pending tx {} has been reverted: {}",
-                    tx_hash, reason
+                    tx_hash,
+                    execution_result.revert_reason().unwrap()
```

Comment:
are we sure this unwrap can't panic? if not we should use `unwrap_or` and give a default value or something, same for below

Reply:
- We are sure that this wont failed, as the status is `Reverted` checked in the match. `revert_reason()` return `None` when the status is not `Reverted`